src/OpenLoco/Objects/VehicleObject.cpp

```diff
@@ -0,0 +1,186 @@
+
+    static loco_global<uint16_t, 0x0052622E> _52622E; // Tick related
+
+    // 0x004B7733
+    static void drawVehicle(Gfx::drawpixelinfo_t* dpi, const vehicle_object* vehicleObject, uint8_t eax, uint8_t esi, Gfx::point_t offset)
+    {
+        // Eventually calls 0x4B777B part of 0x4B7741
+        registers regs;
+        regs.cx = offset.x;
+        regs.dx = offset.y;
+        regs.eax = eax;
+        regs.esi = esi;
+        regs.bl = Colour::saturated_green;
+        regs.bh = 2;
+        regs.ebp = (uintptr_t)vehicleObject;
+        regs.edi = (uintptr_t)dpi;
+        call(0x4B7733, regs);
+    }
+
+    // 0x004B8C52
+    void vehicle_object::drawPreviewImage(Gfx::drawpixelinfo_t& dpi, const int16_t x, const int16_t y) const
+    {
+        Gfx::drawpixelinfo_t* clipped = nullptr;
+
+        xy32 pos = { x - objectPreviewOffset.x, y - objectPreviewOffset.y };
+        if (Gfx::clipDrawpixelinfo(&clipped, &dpi, pos, objectPreviewSize))
+        {
+            // Rotation
+            uint8_t unk1 = _52622E & 0x3F;
+            uint8_t unk2 = ((_52622E + 2) / 4) & 0x3F;
+            drawVehicle(clipped, this, unk1, unk2, { static_cast<int16_t>(objectPreviewOffset.x), 75 });
+        }
+    }
+
+    // TODO: Should only be defined in ObjectSelectionWindow
+    static const uint8_t descriptionRowHeight = 10;
+
+    // 0x004B8C9D
+    void vehicle_object::drawDescription(Gfx::drawpixelinfo_t& dpi, const int16_t x, const int16_t y) const
+    {
+        Gfx::point_t rowPosition = { x, y };
+        ObjectManager::drawGenericDescription(dpi, rowPosition, designed, obsolete);
+        if (power != 0 && (mode == TransportMode::road || mode == TransportMode::rail))
+        {
+            FormatArguments args{};
+            args.push(power);
+            Gfx::drawString_494B3F(dpi, rowPosition.x, rowPosition.y, Colour::black, StringIds::object_selection_power, &args);
+            rowPosition.y += descriptionRowHeight;
+        }
+
+        {
+            FormatArguments args{};
+            args.push(weight);
+            Gfx::drawString_494B3F(dpi, rowPosition.x, rowPosition.y, Colour::black, StringIds::object_selection_weight, &args);
+            rowPosition.y += descriptionRowHeight;
+        }
+        {
+            FormatArguments args{};
+            args.push(speed);
+            Gfx::drawString_494B3F(dpi, rowPosition.x, rowPosition.y, Colour::black, StringIds::object_selection_max_speed, &args);
+        }
+        auto buffer = const_cast<char*>(StringManager::getString(StringIds::buffer_1250));
+
+        getCargoString(buffer);
+
+        // ****TODO: BEFORE MERGE, CLIP DPI PRIOR TO CALLING!!!*****
+        auto width = dpi.width - 4;
+        Gfx::drawString_495224(dpi, rowPosition.x, rowPosition.y, width, Colour::black, StringIds::buffer_1250);
+
+    }
+
+    void vehicle_object::getCargoString(char* buffer) const
+    {
+        if (num_simultaneous_cargo_types != 0)
+        {
+            {
```

Comment:
This scoped block doesn't seem necessary. Is there a reason behind it I'm missing?

Reply:
Its to stop cargoType from being reused.